packages/protocol/contracts/governance/ReleaseGoldInstance.sol

```diff
@@ -0,0 +1,570 @@
+   * @param releaseStartTime The time (in Unix time) at which point releasing starts.
+   * @param releaseCliffTime Duration (in seconds) after `releaseStartTime` of the golds' cliff.
+   * @param numReleasePeriods Number of releasing periods.
+   * @param releasePeriod Duration (in seconds) of each release period.
+   * @param amountReleasedPerPeriod The released gold amount per period.
+   * @param revocable Whether the release schedule is revocable or not.
+   * @param _beneficiary Address of the beneficiary to whom released tokens are transferred.
+   * @param _releaseOwner Address capable of revoking, setting the liquidity provision
+   *                      and setting the withdrawal amount.
+   *                      Null if grant is not subject to these operations.
+   * @param _refundAddress Address that receives refunded funds if contract is revoked.
+   *                       Null if contract is not revocable.
+   * @param subjectToLiquidityProvision If this schedule is subject to a liquidity provision.
+   * @param initialDistributionPercentage Percentage of total rewards available for distribution.
+   *                                      Expressed to 3 significant figures [0, 1000].
+   * @param _canValidate If this schedule's gold can be used for validating.
+   * @param _canVote If this schedule's gold can be used for voting.
+   * @param registryAddress Address of the deployed contracts registry.
+   */
+  constructor(
+    uint256 releaseStartTime,
+    uint256 releaseCliffTime,
+    uint256 numReleasePeriods,
+    uint256 releasePeriod,
+    uint256 amountReleasedPerPeriod,
+    bool revocable,
+    address payable _beneficiary,
+    address _releaseOwner,
+    address payable _refundAddress,
+    bool subjectToLiquidityProvision,
+    uint256 initialDistributionPercentage,
+    bool _canValidate,
+    bool _canVote,
+    address registryAddress
+  ) public {
+    releaseSchedule.numReleasePeriods = numReleasePeriods;
+    releaseSchedule.amountReleasedPerPeriod = amountReleasedPerPeriod;
+    releaseSchedule.releasePeriod = releasePeriod;
+    releaseSchedule.releaseCliff = releaseStartTime.add(releaseCliffTime);
+    releaseSchedule.releaseStartTime = releaseStartTime;
+    require(releaseSchedule.numReleasePeriods >= 1, "There must be at least one releasing period");
+    require(
+      releaseSchedule.amountReleasedPerPeriod > 0,
+      "The released amount per period must be greater than zero"
+    );
+    require(
+      _beneficiary != address(0),
+      "The release schedule beneficiary cannot be the zero addresss"
+    );
+    require(registryAddress != address(0), "The registry address cannot be the zero address");
+    require(
+      releaseSchedule.releaseStartTime.add(
+        releaseSchedule.numReleasePeriods.mul(releaseSchedule.releasePeriod)
+      ) >
+        block.timestamp,
+      "Release schedule end time must be in the future"
```

Comment:
This is left from Limechain's original take on vesting, but I left it because it seems like a decent safeguard against mistakes in provided parameters. Hitting this check would be equivalent to just transferring the granted amount to the beneficiary, with the only downside in transfer being the lack of clear record. 
I don't feel strongly either way, but have a slight bias towards making operator mistakes harder, i.e. making it less probable to provide the wrong start date